telefunc/node/server/shield/codegen/transformer.ts

```diff
@@ -0,0 +1,97 @@
+export { shieldTelefunctions }
+
+import { Project, VariableDeclarationKind } from 'ts-morph'
+
```

Comment:
We can add
```js
const typesSrc = readFileSync(`${__dirname}/types.d.ts`)
```
 here.

`__dirname` doesn't exist in Cloudflare Workers but that's not a problem here.

This module is always loaded in Node.js (in dev and during the build process).

In production this module is not loaded. (None of Vite's code is actually loaded in production.)

Reply:
That doesn't work during test.

Then `__dirname` is:

```
telefunc/telefunc/dist/tests/server/shield/codegen/types.d.ts
```

Reply:
> `types.d.ts` is not present at all

```ts
const telefuncSrcDir = `${__dirname}/../../../../telefunc`
const typesDeclarationFile = `${telefuncSrcDir}/node/server/shield/codegen/types.d.ts`
```
While adding `node/server/shield/codegen/types.d.ts` to `telefunc/pacakge.json#files`.

Or we configure TS to add `types.d.ts` in `dist/` (not sure if that's possible).


> `.d.ts` files don't contain type level types

Not sure what you mean.

Reply:
That works!

This is what I meant: 

`dist/tests`:
![image](https://user-images.githubusercontent.com/649392/164058415-7ccc7c6d-ce06-4a55-adc0-4b420747aa26.png)

`dist/node`:

![image](https://user-images.githubusercontent.com/649392/164058485-5b3a887c-1321-476c-9faf-80adf9b191e7.png)